### PR TITLE
Add example files and update api docs

### PR DESCRIPTION
**Pull request by nlakritz**
- Added FileHeader and CashLetter example files
- Fixed several fields in API docs, many had incorrect type or some other issue

### CODE REVIEW

**Comment by adamdecaf** [comment]
@nlakritz Do you have a spec that was referenced to make this change? There are multiple versions of X9 so I'm not sure where this comes from.

**Comment by nlakritz** [comment]
@adamdecaf @atonks2 I noticed most of the examples in the API documentation return type errors. The [testdata](https://github.com/moov-io/imagecashletter/tree/master/test/testdata) example files work fine though. So, I compared the API examples with the valid testdata ones and found that there were type differences in a few fields. In this PR, I modified the API docs to match to the testdata examples. 

For example, `micrValidIndicator` is a string in the [API documentation](https://moov-io.github.io/imagecashletter/api/#post-/files/-fileID-/cashLetters). Attempting to create a file in that case gives, `{"error":"error creating file from JSON: problem reading file: json: cannot unmarshal string into Go struct field Bundle.cashLetters.bundles.checks of type int"}`.

**Comment by atonks2** [comment]
Oh that makes sense @nlakritz. Initially it looked like this was changing data types, but now I see it seems to be making the documentation match the implementation. I'll review this soon so we can get it merged in.

**Comment by adamdecaf** [comment]
Awesome, thanks!

File: client/docs/CheckDetailAddendumA.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Name | Type | Description | Notes
...
-**BOFDBOFDCorrectionIndicator** | **string** | BOFDCorrectionIndicator identifies whether and how the MICR line of this item was repaired by the creator of this CheckDetailAddendumA Record for fields other than Payor Bank Routing Number and Amount. * &#x60;0&#x60; - No Repair * &#x60;1&#x60; - Repaired (form of repair unknown) * &#x60;2&#x60; - Repaired without Operator intervention * &#x60;3&#x60; - Repaired with Operator intervention * &#x60;4&#x60; - Undetermined if repair has been done or not  | [optional] 
+**BOFDBOFDCorrectionIndicator** | **int32** | BOFDCorrectionIndicator identifies whether and how the MICR line of this item was repaired by the creator of this CheckDetailAddendumA Record for fields other than Payor Bank Routing Number and Amount. * &#x60;0&#x60; - No Repair * &#x60;1&#x60; - Repaired (form of repair unknown) * &#x60;2&#x60; - Repaired without Operator intervention * &#x60;3&#x60; - Repaired with Operator intervention * &#x60;4&#x60; - Undetermined if repair has been done or not  | [optional]
```
Comment (atonks2) [inline client/docs/CheckDetailAddendumA.md:17]: Looks like an existing typo. Should be `BOFDCorrectionIndicator`

**Review by atonks2 — APPROVED** [review]

**Pull request merged**